app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -348,17 +355,103 @@ private void setRoots() {
+            if(record.getStatus() == ApplicationRecord.STATUS_INSTALLED) {
                 return initializeAppResources(new CommCareApp(record));
             }
         }
+        // There are no installed apps
         return STATE_UNINSTALLED;
     }
 
-    private int initializeAppResources(CommCareApp app) {
+    /**
+     * @return all installed ApplicationRecords, in alphabetical order
+     */
+    public ArrayList<ApplicationRecord> getInstalledAppRecords() {
+        ArrayList<ApplicationRecord> records = new ArrayList<>();
+        for (ApplicationRecord r : getGlobalStorage(ApplicationRecord.class)) {
+            records.add(r);
+        }
+        Collections.sort(records, new Comparator<ApplicationRecord>() {
+
+            @Override
+            public int compare(ApplicationRecord lhs, ApplicationRecord rhs) {
+                return lhs.getDisplayName().compareTo(rhs.getDisplayName());
+            }
+
+        });
+        return records;
+    }
+
+    /**
+     * @return all ApplicationRecords that are installed and NOT archived
+     */
+    public ArrayList<ApplicationRecord> getVisibleAppRecords() {
+        ArrayList<ApplicationRecord> visible = new ArrayList<>();
+        for (ApplicationRecord r : getInstalledAppRecords()) {
+            if (!r.isArchived()) {
+                visible.add(r);
+            }
+        }
+        return visible;
+    }
+
+    /**
+     * @return all ApplicationRecords that are installed AND are not archived AND have MM verified
+     */
+    public ArrayList<ApplicationRecord> getReadyAppRecords() {
+        ArrayList<ApplicationRecord> ready = new ArrayList<>();
+        for (ApplicationRecord r : getVisibleAppRecords()) {
+            if (r.resourcesValidated()) {
+                ready.add(r);
+            }
+        }
+        return ready;
+    }
+
+    public boolean visibleAppsPresent() {
+        boolean b = getVisibleAppRecords().size() > 0;
+        return b;
+    }
+
+    /**
+     * @return the list of all installed apps as an array
+     */
+    public ApplicationRecord[] appRecordArray() {
+        ArrayList<ApplicationRecord> appList = CommCareApplication._().getInstalledAppRecords();
+        ApplicationRecord[] appArray = new ApplicationRecord[appList.size()];
+        int index = 0;
+        for (ApplicationRecord r : appList) {
+            appArray[index++] = r;
+        }
+        return appArray;
+    }
+
+    /**
+     *  @return the ApplicationRecord that corresponds to the given index in
+     *  the global list of installed apps
+     */
+    public ApplicationRecord getAppAtIndex(int index) {
+        ArrayList<ApplicationRecord> currentApps = getInstalledAppRecords();
+        if (index < 0 || index >= currentApps.size()) {
+            return null;
+        } else {
+            return currentApps.get(index);
+        }
+    }
+
+    public int initializeAppResources(CommCareApp app) {

```

Comment:
You've added a bunch of methods in between the two overloaded versions of initializeAppResources. Please move the initializeAppResources definitions next to each other, it will make the code easier to read.